### DISCUSSION TITLE
Usage stats

### DISCUSSION THREAD

**Discussion by nixshal** [giswqs/streamlit-geospatial#26] [Q&A] [answered]
Hello,

This is more of a Streamlit question but do you plan to implement usage stats for your Streamlit app? E.g how many times an app was run, what text input was given, how many times a button was clicked, page views... that sort of thing.

I'm curious about how to achieve this on Streamlit and was wondering if you already had something like this in mind.

One option I found was [streamlit-analytics](https://github.com/jrieke/streamlit-analytics), but it didn't work for me using the multi-page method.

**Comment by giswqs** [comment, marked as answer]
This would be a nice feature to add. One potential way is to add Google Analytics into the HTML. I will look into it. 

https://support.google.com/analytics/answer/1008080?hl=en#zippy=%2Cin-this-article
https://www.pair.com/support/kb/how-to-google-analytics-to-an-html-website/

**Reply by nixshal** [reply]
Thank you! Would appreciate a tutorial

**Reply by nixshal** [reply]
Specifically for streamlit, using the multi-page method as used in your web app. Hard to find info on this.

**Comment by giswqs** [comment]
I have added Google Analytics to the web app. See this [commit](https://github.com/giswqs/streamlit-geospatial/commit/e0785313ee61b374d838803b4d28fce0b834dc12) to see what changes I made to `index.html`

**Reply by nixshal** [reply]
Thank you. However, when I tried this on my end the **google-site-verification** doesn't show up when you open your streamlit app and inspect source and look for it.

I couldn't get any data showing up on the GA analytics page too.

**Reply by nixshal** [reply]
I have a feeling a different approach is needed for a private repo
